src/components/DropdownMenu/DropdownMenu.tsx

```diff
@@ -45,107 +58,67 @@ export const DropdownMenu: React.FC<Props> = ({
   children,
   className,
   isActive,
-  handleOnKeyDown,
+  handleOnEscDown,
   ...other
 }) => {
-  const childRefs = useRef<Array<HTMLLIElement>>([]);
+  const refs = useRef<Refs>({
+    set: new Set(),
+    list: [],
+  });
+
+  let focusIndex = 0;
+  const focusItem = (index: number) => {
+    refs.current.list[index]
+      ?.querySelector<HTMLButtonElement | HTMLAnchorElement>(':first-child')
+      ?.focus();
+  };
 
   useEffect(() => {
-    setTimeout(() => {
-      if (isActive && childRefs) {
-        childRefs.current[0]
-          ?.querySelector<HTMLButtonElement | HTMLAnchorElement>(':first-child')
-          ?.focus();
-      }
-    }, 1);
```

Comment:
Successfully focuses first element without this setTimeout, also reflected in unit test